types/didyoumean/index.d.ts

```diff
@@ -0,0 +1,26 @@
+// Type definitions for didyoumean 1.2
+// Project: https://github.com/dcporter/didyoumean.js
+// Definitions by: James George <https://github.com/jamesgeorge007>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.7
+
+export = didYouMean;
+
+declare function didYouMean(str: string, list: string[], key?: string): string | string[];
+
+declare namespace didYouMean {
+    let caseSensitive: boolean;
+
+    let nullResultValue: any;
+
+    let prototype: {
```

Comment:
I don't believe you need to explicitly declare `prototype` - you should get it from the merged function declaration.